app/models/storage.rb

```diff
@@ -18,7 +18,6 @@ class Storage < ApplicationRecord
   has_many :storage_files,       :dependent => :destroy
   has_many :storage_files_files, -> { where("rsc_type = 'file'") }, :class_name => "StorageFile", :foreign_key => "storage_id"
   has_many :files,               -> { where("rsc_type = 'file'") }, :class_name => "StorageFile", :foreign_key => "storage_id"
-  has_many :host_storages
```

Comment:
What is this doing here?  Seems unrelated to the original PR.

Reply:
Sorry, let me explain.  There's two `host_storages` has_manys.  One on line line 5 and one here, at line 21.  Because this one on line 21 is "last", it wins.  This causes a problem because line 6, we have:

`has_many :hosts,             :through => :host_storages`

This causes complaints on rails 5.1 because the through association is needed before it's defined.